### PR TITLE
[IMP] wowl: dialog component (prototype)

### PR DESCRIPTION
**Pull request by Polymorphe57**

### CODE REVIEW

File: addons/wowl/static/tests/components/dialog_tests.ts [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    static template = owl.tags.xml`
+            <Dialog fullscreen="true"/>
+        `;
+  }
+  await mount(Parent, { env, target });
+  assert.containsOnce(target, ".o_dialog");
+  assert.hasClass(target.querySelector(".o_dialog .modal") as HTMLElement, "o_modal_full");
+});
+
+QUnit.skip("Interactions between multiple dialogs", async function (assert) {
```
Comment (ged-odoo) [inline addons/wowl/static/tests/components/dialog_tests.ts]: skippé

File: addons/wowl/static/src/components/dialog/dialog.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    title: "Odoo",
+  };
+  static props: {
+    fullscreen: boolean;
+    renderFooter: boolean;
+    renderHeader: boolean;
+    size: {
+      type: String;
+      validate: (s: string) => true;
+      // validate: (s: string) => ["modal-xl", "modal-lg", "modal-sm"].includes(s),
```
Comment (Polymorphe57) [inline addons/wowl/static/src/components/dialog/dialog.ts]: does not work: Typescript complains...

File: addons/wowl/static/src/components/dialog/dialog.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import { Component, hooks, misc } from "@odoo/owl";
+const { useRef, useExternalListener } = hooks;
+const { Portal } = misc;
+
+export class Dialog extends Component {
+  static components = { Portal };
+  static props: {
```
Comment (ged-odoo) [inline addons/wowl/static/src/components/dialog/dialog.ts]: `static props = {`

File: addons/wowl/static/src/components/dialog/dialog.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+  constructor() {
+    super(...arguments);
+    useExternalListener(window, "keydown", this._onKeydown);
+  }
+
+  mounted() {
+    const dialogContainer = document.querySelector(".o_dialog_container") as HTMLElement;
+    const modals = dialogContainer.querySelectorAll(".o_dialog .modal");
+    const len = modals.length;
```
Comment (ged-odoo) [inline addons/wowl/static/src/components/dialog/dialog.ts:45]: much simpler to simply do 
```js
for (let model of modals) {
      modal.classList.add("o_inactive_modal");
}
```

Comment (ged-odoo) [reply addons/wowl/static/src/components/dialog/dialog.ts:45]:
```js
for (let model of modals.slice(-1)) {
      modal.classList.add("o_inactive_modal");
}
```

Comment (Polymorphe57) [reply addons/wowl/static/src/components/dialog/dialog.ts:45]: as discussed, we won't do that

**Review by ged-odoo — APPROVED** [review]

**Pull request merged**